huaweicloud/services/rds/resource_huaweicloud_rds_pg_account.go

```diff
@@ -0,0 +1,342 @@
+	}
+
+	// Split instance_id and user from resource id
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return diag.Errorf("invalid ID format, must be <instance_id>/<name>")
+	}
+	instanceId := parts[0]
+	accountName := parts[1]
+
+	getPgAccountPath := getPgAccountClient.Endpoint + getPgAccountHttpUrl
+	getPgAccountPath = strings.ReplaceAll(getPgAccountPath, "{project_id}", getPgAccountClient.ProjectID)
+	getPgAccountPath = strings.ReplaceAll(getPgAccountPath, "{instance_id}", instanceId)
+
+	getPgAccountResp, err := pagination.ListAllItems(
+		getPgAccountClient,
+		"page",
+		getPgAccountPath,
+		&pagination.QueryOpts{MarkerField: ""})
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving RDS PostgreSQL account")
+	}
+
+	getPgAccountRespJson, err := json.Marshal(getPgAccountResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var getPgAccountRespBody interface{}
+	err = json.Unmarshal(getPgAccountRespJson, &getPgAccountRespBody)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	account := utils.PathSearch(fmt.Sprintf("users[?name=='%s']|[0]", accountName), getPgAccountRespBody, nil)
+
+	if account == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", instanceId),
+		d.Set("name", utils.PathSearch("name", account, nil)),
+		d.Set("description", utils.PathSearch("comment", account, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourcePgAccountUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updatePgAccountClient, err := cfg.NewServiceClient("rds", region)
+	if err != nil {
+		return diag.Errorf("error creating RDS client: %s", err)
+	}
+
+	if err = updatePgAccountPassword(ctx, d, updatePgAccountClient); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err = updatePgAccountDescription(d, updatePgAccountClient); err != nil {
+		return diag.FromErr(err)
+	}
+	return resourcePgAccountRead(ctx, d, meta)
+}
+
+func updatePgAccountPassword(ctx context.Context, d *schema.ResourceData, client *golangsdk.ServiceClient) error {
+	if !d.HasChange("password") {
+		return nil
+	}
+	// updatePgAccount: update RDS PostgreSQL account password
+	updatePgAccountHttpUrl := "v3/{project_id}/instances/{instance_id}/db_user/resetpwd"
+
+	updatePgAccountPath := client.Endpoint + updatePgAccountHttpUrl
+	updatePgAccountPath = strings.ReplaceAll(updatePgAccountPath, "{project_id}", client.ProjectID)
+	updatePgAccountPath = strings.ReplaceAll(updatePgAccountPath, "{instance_id}", d.Get("instance_id").(string))
+
+	updatePgAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	updatePgAccountOpt.JSONBody = utils.RemoveNil(buildUpdatePgAccountPasswordBodyParams(d))
+
+	instanceId := d.Get("instance_id").(string)
```

Comment:
Repeat field definition. I think this line can move to 219.

Reply:
done